### DISCUSSION TITLE
Dealing with logging bots & staff

### DISCUSSION THREAD

**Discussion by ezra-351** [victornpb/undiscord#441] [2. Discussions]
Let's say, purely hypothetically, I'm in a server that's been down the drain for years, and the staff team insists on necromancing until the end of time. They're not nice people, and if they could find any reason to ban me, for, say breaking TOS, they would. They have a bot for logging deleted messages, fortunately the bulk of the ~24k messages to delete were sent before the bot was implemented and could cache them, so it would simply log something like "Unknown message deleted in \#channel".

And for the sake of argument, I was aware of this & banked on staff having the channel muted, and wouldn't need to check it until I was done. Murphy's law, roughly 3 hours & ~4k messages later, an admin peeks at the log and panics at thousands of messages being deleted for seemingly no reason. The staff temporarily lock down the server, stopping me from being able to delete messages. Without checking audit logs, they collectively assume it's the bot having a meltdown, kick and replace it with a new one, badabing badaboom problem solved let's move on and forget this happened.

In this scenario, my worry would be that if they notice it starts happening again, start digging deeper and check audit logs, they'd ban me on the accusation of breaking TOS (something i would never do ofc ofc), leaving my messages there for as long as they keep hauling this corpse of a server away from its grave.

I understand this is a very hyperspecific scenario, so I encourage this discussion to branch out within the regards of scrubbing messages in servers that keep logs and/or have staff that would disallow that, etc.

**Comment by victornpb** [comment]
Technically doing that is against laws like GDPR, the server owners would have disclose that they are keeping your data, so they would be also have to comply to things like, right to be forgotten, have an explicit privacy policy.

I'm no expert on the subject, but realistically I know many servers do that, and no one does anything about it, so I'm not sure what we could do about it. You could ask Discord support, but we know how little they care #429
